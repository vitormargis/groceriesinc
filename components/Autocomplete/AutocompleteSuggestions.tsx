import React, { FC, MouseEvent } from 'react';

interface AutocompleteOptionsProps {
  active: number,
  filtered: string[],
  onMouseEnter: (e: MouseEvent) => void, 
  onClick: (e: MouseEvent) => void
}

const AutocompleteOptions: FC<AutocompleteOptionsProps> = ({active, filtered, onMouseEnter, onClick})  => {
  return filtered.length ? (
    <ul className="autocomplete-suggestions" data-cy="autocomplete-suggestions">
      {filtered.map((suggestion, index) => {
        return (
          <li 
            className={index === active ? 'active' : ''} 
            data-cy={`autocomplete-suggestion-${index}`} 
            key={suggestion} 
            onClick={onClick} 
            onMouseEnter={onMouseEnter}>
            {suggestion}
          </li>
        );
      })}
    </ul>
  ) : null
}

export default AutocompleteOptions; 