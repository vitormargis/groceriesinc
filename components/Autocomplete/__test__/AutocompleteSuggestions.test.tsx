import React from 'react';
import { act, create } from 'react-test-renderer';
import AutocompleteSuggestions from '../AutocompleteSuggestions';

const props = {
  onClick: () => null,
  active: 0,
  filtered: [ 'apple' ],
  onMouseEnter: () => null
}

const renderComponent = (props: any) => {
  let result: any;
  act(() => {
    result = create(<AutocompleteSuggestions {...props} />);
  });

  return result;
};

describe('AutocompleteSuggestions', () => {
  it('should render with AutocompleteSuggestions', () => {
    const component = renderComponent(props);
    component.root.findByType('li').props.onClick();
    expect(component.toJSON).toMatchSnapshot();
  });
});