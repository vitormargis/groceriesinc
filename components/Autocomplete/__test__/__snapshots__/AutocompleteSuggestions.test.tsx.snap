// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteSuggestions should render with AutocompleteSuggestions 1`] = `[Function]`;
